lib/rspec/rails/view_rendering.rb

```diff
@@ -39,18 +39,50 @@ def render_views?
         self.class.render_views? || !controller.class.respond_to?(:view_paths)
       end
 
-      # Delegates find_all to the submitted path set and then returns templates
-      # with modified source
-      #
-      # @private
-      class EmptyTemplateResolver < ::ActionView::FileSystemResolver
+      class EmptyTemplateResolverFactory
+        def initialize(path)
+          @path = path
+        end
 
-        def initialize(path, pattern=nil)
-          unless path.is_a?(::ActionView::Resolver)
-            super
+        def resolver
+          if @path.is_a?(::ActionView::Resolver)
+            EmptyTemplateResolverDecorator.new(@path)
+          else
+            EmptyTemplateFileSystemResolver.new(@path)
           end
         end
+      end
 
+      class EmptyTemplateResolverDecorator
+        def initialize(resolver)
+          @resolver = resolver
+        end
+
+        def method_missing(name, *args, &block)
+          @resolver.send(name, *args, &block)
+        end
+
+      private
+
+        def find_templates(*args)
+          templates = @resolver.find_templates(*args)
+          templates.map do |template|
```

Comment:
This map could probably now be delegated to outside as it does the same as below.